### PR TITLE
New Alert Card component & deprecation of Alerts Table

### LINKED ISSUE #1397: Create Alert Card & Deprecate Alerts Table
### Description

Due to lack of real estate, the alerts table will be getting replaced with alerts cards. The alerts cards will share similar information to the currently visible alerts table, but will allow us to easily extend it & add more values

### Designs

https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/849670d3-62c0-462f-b399-bf6174c4352b/commits/latest/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/FA1F7045-29D1-4B29-98FC-AD10075AACE1?collectionId=ef90166e-2279-4166-af63-e4b40b4f545c&collectionLayerId=b3aa0c9b-cbbd-4dfe-9f90-64c212b0f97c

### Acceptance Criteria

- An `AlertCard` can be created using the `GenericItemCard` component, following the `DestinationCard` principles
- The `AlertCard` is tested for displaying the correct information
- Alerts table is no longer present, but gets replaced with Alert Cards in the following pages:
  - ListAlertsPage
  - LogAnalysisOverviewPage
  - RuleDetailsPage

### Notes

- Destinations won't be shown in the `AlertCard` (not currently returned by the API)
- Log Types won't be shown in the `AlertCard` (not currently shown by the API)

### PR DESCRIPTION
**Pull request by alexmylonas**
## Description

This PR introduces a new component `AlertCard` to replace all instances of AlertsTable we are using. More specifically this card will be used on `ListAlertsTable`, `LogAnalysisOverview` and `RuleDetailsPage`

This is considered a better way to display alerts on all those places.

You can find an updated mock [here](https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/849670d3-62c0-462f-b399-bf6174c4352b/commits/latest/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/FA1F7045-29D1-4B29-98FC-AD10075AACE1?collectionId=ef90166e-2279-4166-af63-e4b40b4f545c&collectionLayerId=b3aa0c9b-cbbd-4dfe-9f90-64c212b0f97c)

Design notes: 
- Destinations won't be shown in the AlertCard (not currently returned by the API)
- Log Types are included

Closes #1397, #1398

**Important note**: With this change, the ability to sort Alerts based on severity, creation time etc is disabled.

### Alerts Page
Screenshot 2020-09-15 at 4 42 35 PM

### Rules Details (with alerts)

Screenshot 2020-09-15 at 4 42 28 PM

### Log Analysis Overview page
Screenshot 2020-09-15 at 4 24 42 PM

### Checklist:

- [x] Make sure you give an high-level overview of the problem & the solution
- [x] Make sure the related issue is referenced
- [x] Make sure screenshots are added if the UI is changed in any way
- [x] Make sure all of the significant logic is covered by tests
- [x] Make sure your changes are accessible (addition of aria-attributes & correct HTML semantics)
- [x] Make sure you reference the related design document if it exists
- [x] Make sure the changes are responsive (appearing decently on devices)
- [x] ~~Make sure telemetry is added (analytics & errors)~~

### CODE REVIEW

File: web/src/components/GenericItemCard/GenericItemCard.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ interface GenericItemCardLogoProps {
 
 interface GenericItemCardValueProps {
   label: string;
-  value: string | React.ReactElement;
+  value: string | number | React.ReactElement;
```
Comment (alexmylonas) [inline web/src/components/GenericItemCard/GenericItemCard.tsx:29]: To cover `Events` which are integers

File: web/src/components/cards/AlertCard/AlertCard.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        <GenericItemCard.ValuesGroup>
+          <Link external as={RRLink} mt={4} to={urls.logAnalysis.rules.details(alert.ruleId)}>
+            <Button variantColor="navyblue" as="div">
+              <Text fontSize="small">View Rule</Text>
+            </Button>
+          </Link>
+          <GenericItemCard.Value label="Events" value={alert.eventsMatched} />
+          <GenericItemCard.Value
+            label="Time Created"
+            value={formatDatetime(alert.creationTime, true)}
```
Comment (alexmylonas) [inline web/src/components/cards/AlertCard/AlertCard.tsx]: On mocks for this, there seems to be more margin between label and value, should I changed in the `GenericItemCard.Value` component

File: web/src/components/cards/AlertCard/index.tsx [hunk 3]
```diff
@@ -16,4 +16,4 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-export { default } from './AlertsTable';
+export { default } from './AlertCard';
```
Comment (alexmylonas) [inline web/src/components/cards/AlertCard/index.tsx:19]: Git wrongly understands this as replacement, its a new component `AlertCard` and deletion of `AlertsTable`.

Also, I took the liberty of creating a new subfolder on `components` named `cards`

File: web/src/pages/ListAlerts/ListAlerts.tsx [hunk 4]
```diff
@@ -20,23 +20,21 @@ import React from 'react';
 import { Alert, Box, Card } from 'pouncejs';
 import { DEFAULT_LARGE_PAGE_SIZE } from 'Source/constants';
 import { extractErrorMessage } from 'Helpers/utils';
-import { ListAlertsInput, SortDirEnum, ListAlertsSortFieldsEnum } from 'Generated/schema';
```
Comment (alexmylonas) [inline web/src/pages/ListAlerts/ListAlerts.tsx:23]: The removal of sorting the alerts 😕

**Comment by jacknagz** [comment]
@alexmylonas do we know when the destinations/log types will be available in the API?

**Comment by alexmylonas** [comment]
> @alexmylonas do we know when the destinations/log types will be available in the API?

@jacknagz, @s0l0ist just made merged a PR for LogTypes (https://github.com/panther-labs/panther/pull/1544) so they will be included on this PR after a small change.

Destinations has a bit more work to my understanding and we'll be pushed for a next release!

**Comment by jacknagz** [comment]
@alexmylonas awesome, thanks for the heads up! Is @s0l0ist owning the destinations part? Do we have a follow-up ticket to track?

**Comment by s0l0ist** [comment]
> @alexmylonas awesome, thanks for the heads up! Is @s0l0ist owning the destinations part? Do we have a follow-up ticket to track?

@jacknagz the destinations part should be trivial to implement once #1497 has been merged. I can own this little part.

File: web/src/pages/ListAlerts/ListAlerts.tsx (hunk 4 above)
Comment (3nvi) [reply web/src/pages/ListAlerts/ListAlerts.tsx:23]: Let's sync with our designer about it (as we agreed)

File: web/src/pages/RuleDetails/RuleDetails.tsx [hunk 5]
```diff
@@ -129,9 +128,12 @@ const RuleDetailsPage = () => {
       <Box mt={5} mb={6}>
         <Panel title="Alerts">
           <ErrorBoundary>
-            {hasAnyAlerts && (
-              <RuleDetailsAlertsTable alerts={listAlertsData.alerts.alertSummaries} />
-            )}
+            {hasAnyAlerts &&
+              listAlertsData.alerts.alertSummaries.map((alert, i) => (
```
Comment (3nvi) [inline web/src/pages/RuleDetails/RuleDetails.tsx:132]: we could switch with:

```tsx
<Flex direction="column" spacing={2}>
{alertItems.map(alert => <AlertCard key={alert.alertId} alert={alert} />)}
</Flex>
```

This way you don't have the extra margin at the first/last items 😉

**Review by 3nvi — COMMENTED** [review]
Really good work @alexmylonas . Some minor changes and we are g2g 🎉

File: web/src/components/cards/AlertCard/AlertCard.test.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import AlertCard from './index';
+
+describe('AlertCard', () => {
+  it('displays Alert data in the card', async () => {
+    const alertData = buildAlertSummary();
+
+    const { getByText, getByAriaLabel } = render(<AlertCard alert={alertData} />);
+
+    expect(getByText(alertData.title)).toBeInTheDocument();
+    expect(getByText('View Rule')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (alexmylonas) [reply web/src/components/cards/AlertCard/AlertCard.test.tsx:32]: Add new test check links for Rule link and Alert link!

File: web/src/components/cards/AlertCard/AlertCard.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+interface AlertCardProps {
+  alert: AlertSummaryFull;
+}
+
+const AlertCard: React.FC<AlertCardProps> = ({ alert }) => {
+  return (
+    <GenericItemCard>
+      <GenericItemCard.Body>
+        <Link
+          external
```
[the remark this answers is not in the document]
Comment (alexmylonas) [reply web/src/components/cards/AlertCard/AlertCard.tsx]: Removed `external`

File: web/src/components/cards/AlertCard/AlertCard.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+          as={RRLink}
+          to={urls.logAnalysis.alerts.details(alert.alertId)}
+          cursor="pointer"
+        >
+          <GenericItemCard.Heading>{alert.title}</GenericItemCard.Heading>
+        </Link>
+        <GenericItemCard.ValuesGroup>
+          <Link external as={RRLink} mt={4} to={urls.logAnalysis.rules.details(alert.ruleId)}>
+            <Button variantColor="navyblue" as="div">
```
[the remark this answers is not in the document]
Comment (alexmylonas) [reply web/src/components/cards/AlertCard/AlertCard.tsx]: Done!

File: web/src/components/cards/AlertCard/AlertCard.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+              <Text fontSize="small">View Rule</Text>
```
[the remark this answers is not in the document]
Comment (alexmylonas) [reply web/src/components/cards/AlertCard/AlertCard.tsx]: Yeap that did it!

File: web/src/pages/RuleDetails/RuleDetails.tsx (hunk 5 above)
Comment (alexmylonas) [reply web/src/pages/RuleDetails/RuleDetails.tsx]: Thanks for this, I will keep it in mind!

**Comment by alexmylonas** [comment]
@3nvi Addressed all your comments. Re-review when you can!

File: web/src/components/cards/AlertCard/AlertCard.test.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    expect(getByText('Time Created')).toBeInTheDocument();
+    expect(getByText(SeverityEnum.Medium)).toBeInTheDocument();
+    expect(getByText(AlertStatusesEnum.Triaged)).toBeInTheDocument();
+    expect(getByAriaLabel('Change Alert Status')).toBeInTheDocument();
+  });
+
+  it('should check links are valid', async () => {
+    const alertData = buildAlertSummary();
+    const { getByText } = render(<AlertCard alert={alertData} />);
+    expect(getByText(alertData.title).closest('a')).toHaveAttribute(
```
Comment (3nvi) [inline web/src/components/cards/AlertCard/AlertCard.test.tsx:43]: I'm ok with this but @vorillaz might not be 😄 

Talking about `.closest('a')`

File: web/src/pages/ListAlerts/ListAlerts.tsx (hunk 4 above)
Comment (3nvi) [reply web/src/pages/ListAlerts/ListAlerts.tsx:23]: What did we agree on this @alexmylonas and @dimitriszoitas ?

File: web/src/pages/LogAnalysisOverview/LogAnalysisOverview.tsx [hunk 11]
```diff
@@ -89,7 +89,15 @@ const LogAnalysisOverview: React.FC = () => {
       </SimpleGrid>
       <SimpleGrid columns={1} spacingX={3} spacingY={2}>
         <Panel title="Recent High Severity Alerts">
-          {loadingAlerts ? <TablePlaceholder /> : <AlertsTable items={alertItems} />}
+          {loadingAlerts ? (
+            <TablePlaceholder />
+          ) : (
+            alertItems.map((alert, i) => (
```
Comment (3nvi) [inline web/src/pages/LogAnalysisOverview/LogAnalysisOverview.tsx:95]: can we follow the same `<Flex>` convention as in above or below file?

**Review by 3nvi — COMMENTED** [review]
@alexmylonas looks good. Minor ccomments mainly.

The main one is sorting out the "sort functionality"

File: web/src/components/cards/AlertCard/AlertCard.test.tsx (hunk 10 above)
Comment (alexmylonas) [reply web/src/components/cards/AlertCard/AlertCard.test.tsx]: Talk with theo, changed this to get link from `aria-label`

File: web/src/pages/LogAnalysisOverview/LogAnalysisOverview.tsx (hunk 11 above)
Comment (alexmylonas) [reply web/src/pages/LogAnalysisOverview/LogAnalysisOverview.tsx]: Done!

File: package.json [hunk 12]
```diff
@@ -120,7 +120,7 @@
     "history": "^4.10.1",
     "linkifyjs": "^2.1.9",
     "lodash": "^4.17.19",
-    "pouncejs": "^0.66.2",
+    "pouncejs": "^0.67.3",
```
Comment (vorillaz) [inline package.json:123]: I think the current pounce version is `0.69`.

**Review by vorillaz — APPROVED** [review]
Nice one 🚨

**Review by 3nvi — APPROVED** [review]
LGTM. Can merge 🎉

**Pull request merged**